src/Apps/Order2/Routes/Checkout/Components/OfferStep/Order2OfferStep.tsx

```diff
@@ -104,6 +108,22 @@ export const Order2OfferStep: React.FC<Order2OfferStepProps> = ({ order }) => {
             orderData.currencyCode,
           )}`
 
+      // Unset the current fulfillment option if it exists
+      if (orderData.selectedFulfillmentOption?.type) {
+        const unsetFulfillmentOptionResult =
+          await unsetOrderFulfillmentOption.submitMutation({
+            variables: {
+              input: {
+                id: orderData.internalID,
+              },
+            },
+          })
+        validateAndExtractOrderResponse(
+          unsetFulfillmentOptionResult.unsetOrderFulfillmentOption
+            ?.orderOrError,
+        )
+      }
+
       const response: useOrder2AddInitialOfferMutation$data =
         await submitOfferMutation({
```

Comment:
non-blocking: We could user the new update offer mutation if the last offer exists.